app/src/main/java/com/android/streetworkapp/ui/train/CircularTimer.kt

```diff
@@ -0,0 +1,69 @@
+package com.android.streetworkapp.ui.train
+
+import androidx.compose.animation.core.Animatable
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.StrokeCap
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.streetworkapp.ui.theme.ColorPalette.INTERACTION_COLOR_DARK
+import com.android.streetworkapp.ui.theme.ColorPalette.PRIMARY_TEXT_COLOR
+import com.android.streetworkapp.ui.theme.ColorPalette.PRINCIPLE_BACKGROUND_COLOR
+import kotlinx.coroutines.delay
+
+/**
+ * The CircularTimer composable displays a circular timer that counts down from a specified time.
+ */
+@Composable
+fun CircularTimer(totalTime: Int = 30, onTimeUp: () -> Unit = {}) {
+  var timeRemaining by remember { mutableStateOf(totalTime) }
+  val progress = remember { Animatable(1f) }
+  var isTimeUp by remember { mutableStateOf(false) }
+
+  LaunchedEffect(key1 = timeRemaining) {
+    if (timeRemaining > 0) {
+      progress.animateTo(
+          targetValue = timeRemaining / totalTime.toFloat(),
+          animationSpec = tween(durationMillis = 1000) // Smooth animation for progress
+          )
```

Comment:
It could be interesting to leave the smoothness of the animation as a parameter, for instance when running on the emmulator the circle counter does not move and then changes all of a sudden each second. This could allow customisation :)

Reply:
I did smooth the animation

Reply:
Nice, it is way smoother now !